server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -129,6 +130,32 @@ private static StorageType fromString(String string) {
     @Nullable // if any snapshot UUID will do
     private String snapshotUuid;
 
+    /**
+     * Alias write index policy for controlling how writeIndex attribute is handled during restore
+     *
+     * @opensearch.api
+     */
+    @PublicApi(since = "3.0.0")
+    public enum AliasWriteIndexPolicy {
+        PRESERVE,
+        STRIP_WRITE_INDEX,
+        CUSTOM_SUFFIX;
```

Comment:
This is doing two things, stripping the write index flag and appending a suffix, I think the writeIndexPolicy should only manipulate that flag and we add suffix with another param, wdyt?